src/react-components/src/components/01-atoms/Forms/FormDropdown.tsx

```diff
@@ -15,34 +17,80 @@ export interface FormDropdownProps {
   onSelectChange: (event: React.FormEvent) => void;
 }
 
-export default function FormDropdown(props: FormDropdownProps) {
-  const { dropdownId, blockName, labelText, options, ariaLabel, disabled, selectedOption, onSelectBlur, onSelectChange } = props;
-  if (!labelText && !ariaLabel) {
-      // TODO: Assign aria-labeledBy to labelText
-      throw new Error("Either labelText or ariaLabel must be defined");
+export default class FormDropdown extends React.Component<FormDropdownProps, { selectedOption: string }> {
+  constructor(props: FormDropdownProps) {
+    super(props);
+    this.state = { selectedOption: "" };
+    this.onSelectChange.bind(this);
   }
 
-  let formItemBlockName = blockName ? blockName : "form-item";
-  let labelClassName = bem("label", ["textField"], formItemBlockName);
-
-  return (
-    <div className={bem("dropdown", [], formItemBlockName)}>
-      {labelText && (<label htmlFor={dropdownId} className={labelClassName}>{labelText}</label>)}
-
-      <select id={dropdownId}
-        className={bem("select", [], formItemBlockName)}
-        value={selectedOption}
-        onChange={onSelectChange}
-        onBlur={onSelectBlur}
-        aria-label={ariaLabel}
-        disabled={disabled ? disabled : false}
-      >
-          { options.map((child, key) => {
-            return <option key={key.toString()} aria-selected={child === selectedOption} value={child}>{ child }</option>;
-          }) }
-      </select>
-
-      <Icon decorative={true} name={"arrow"} modifiers={["small"]} />
-    </div>
-  );
+  componentDidMount() {
+    console.log("mounted props", this.props);
+    this.setState({ selectedOption: this.props.selectedOption });
```

Comment:
I _think_ the right way to set state from props is in the constructor and not in `componentDidMount`, unless if this is now the better way to do this. But, I think it makes more sense to do it in the constructor to not trigger a rerender when the component loads.

Reply:
Yes it does!  I think it's still a good rule of thumb to not put state setting in stuff like `componentDidMount` and my brain just goofed.